M2.el

```diff
@@ -320,8 +320,16 @@ can be executed with \\[M2-send-to-program]."
 	((save-excursion
 	   (beginning-of-line)
 	   ;; example:      ../../m2/res.m2:210:45-214:6: --source code:
-	   ;;                (1                     1)   (2       2)   (3      3)   (4      4)   (5      5)   (6      6))
-	   (looking-at "^ *\\(o+[1-9][0-9]* = \\|| \\)?\\([^:\n]+\\):\\([0-9]+\\):\\([0-9]+\\)-\\([0-9]+\\):\\([0-9]+\\):"))
+	   ;; we also match lines beginning with
+	   ;; * output line numbers, e.g.,:
+	   ;;   o1 = ../../m2/res.m2:210:45-214:6:
+	   ;; * a pipe, for paths enclosed in a box, e.g.,:
+	   ;;   | ../../m2/res.m2:210:45-214:6:
+	   ;; * comments, possibly following additional text, e.g.,:
+	   ;;   foo -- ../../m2/res.m2:210:45-214:6:
+	   ;;   foo -* ../../m2/res.m2:210:45-214:6:
+	   ;;                (1                                         1)   (2       2)   (3      3)   (4      4)   (5      5)   (6      6))
+	   (looking-at "^ *\\(o+[1-9][0-9]* = \\|| \\|.*-- *\\|.*-\\* *\\)?\\([^:\n]+\\):\\([0-9]+\\):\\([0-9]+\\)-\\([0-9]+\\):\\([0-9]+\\):"))
```

Comment:
I don't think this is good `.*-- *`. Did you test it?

Reply:
It probably matches the right things, but I tihnk it might match unexpected things. Note that there's no `.*` in the previous regexp.

Reply:
> I don't think this is good `.*-- *`. Did you test it?

Yeah -- I haven't found any issues yet.

> It probably matches the right things, but I tihnk it might match unexpected things. Note that there's no `.*` in the previous regexp.

If the goal is to match:

```m2
** anything that might go before a comment ** -- /usr/share/Macaulay2/Core/matrix1.m2:669:20-673:21:
```
then it seems like the correct thing to do.  We haven't used it before because we've been very particular about what could go before the path:

* spaces only
* output line numbers
* `|` for boxes

Note that we currently (and still will have) false positives, because the regex for the path itself matches anything except `:` and `\n`, which is probably correct since there's very little restriction on what characters can be used for filenames on most systems.  For example, before this PR:

```m2
  --   /usr/share/Macaulay2/Core/matrix1.m2:669:20-673:21: --source code:
```
brings up a `File not found:  --   /usr/share/Macaulay2/Core/matrix1.m2` error.

Reply:
I reverted this part of the change in #9, because my Emacs hangs on long strings. We can revert #9 whenever that is figured out, but since this is a new feature, doesn't hurt to wait till 1.18.